extension/sigv4authextension/README.md

```diff
@@ -0,0 +1,51 @@
+# Authenticator - Sigv4
+
+This extension provides the Sigv4 process of adding authentication information to AWS API requests to HTTP based exporters for AWS services. This extension implements the `configauth.ClientAuthenticator` interface. For more information on the Sigv4 process, please look [here](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).
+
+The authenticator type has to be set to `sigv4auth`. 
+
+## Configuration
+
+The configuration fields are as follows:
+
+* `region`: **Required**. The AWS region for AWS Sigv4
+    * Note that an attempt will be made to obtain a valid default from the `AWS_DEFAULT_REGION` environment variable
+    * [List of AWS regions](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html)
+* `service`: **Required**. The AWS service for AWS Sigv4
+* `role_arn`: **Optional**. The Amazon Resource Name (ARN) of a role to assume
+* `sts_endpoint`: **Optional**. The endpoint for the `role_arn`
+
+
+```yaml
+extensions:
+  sigv4auth:
+    region: "us-west-2" # replace with your own region
+    service: "aps" # replace with desired service
+    role_arn: "arn:aws:iam::123456789012:role/aws-service-role/access"
+    sts_endpoint: "regional" # accepted values are "legacy" or "regional"
+
+receivers:
+  hostmetrics:
+    scrapers:
+      memory:
+
+exporters:
+  prometheusremotewrite/withauth:
+    endpoint: "https://aps-workspaces.us-west-2.amazonaws.com/workspaces/ws-XXX/api/v1/remote_write"
+    auth:
+      authenticator: sigv4auth
+
+service:
+  extensions: [sigv4auth]
+  pipelines:
+    metrics:
+      receivers: [hostmetrics]
+      processors: []
+      exporters: [prometheusremotewrite/withauth]
+```
+
+## Notes
+
+* You must use your `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as an environment variable in the configuration yaml to provide AWS permissions for your Collector
```

Comment:
There are many ways to setup auth for an AWS SDK, let's just leave this generic like `The collector must have valid AWS credentials as used by the [AWS SDK for Go](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials)`